src/placement/pl_map.c

```diff
@@ -375,6 +375,41 @@ pl_htable_init()
 					   &pl_hash_ops, &pl_htable);
 }
 
+void
+pl_map_destroy_v2(struct pl_map **pl_mapp)
+{
+	if (*pl_mapp != NULL) {
+		struct pl_map	*map = *pl_mapp;
+		bool		 last;
+
+		*pl_mapp = NULL;
+		D_SPIN_LOCK(&map->pl_lock);
+		/* Drop the reference when create */
+		map->pl_ref--;
+		last = (map->pl_ref == 0);
+		D_SPIN_UNLOCK(&map->pl_lock);
+		if (last)
+			pl_map_destroy(map);
+	}
+}
+
+int
+pl_map_create_v2(struct pool_map *pool_map, struct pl_map **pl_mapp)
+{
+	struct pl_map		*pl_map = NULL;
+	struct pl_map_init_attr	 mia;
+	int			 rc;
+
+	pl_map_attr_init(pool_map, PL_TYPE_RING, &mia);
+	rc = pl_map_create_inited(pool_map, &mia, &pl_map);
+	if (rc == 0) {
+		pl_map_destroy_v2(pl_mapp);
+		*pl_mapp = pl_map;
+	}
+
+	return rc;
+}
```

Comment:
Could you use pl_map_create/destroy() directly to create the placement map inside hash?

Reply:
Currently, the pl_map hash table is used by client-stack based logic. It is not available for DTX usage. I will study how to unify them. But since this patch has been merged already, even if they can be unified, it will be in another separated patch.